parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -1452,10 +1557,18 @@ lbrace_cmd_block_start:
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | arg relop error %prec tGT
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
                 | rel_expr relop arg %prec tGT
                     {
                       $$ = driver.build.binaryOp(self, $1, $2, $3);
                     }
+                | rel_expr relop error %prec tGT
+                    {
+                      $$ = driver.build.binaryOp(self, $1, $2, driver.build.error_node(self, @2.end, @3.end));
+                    }
```

Comment:
What are these rules? They don't look like the other cases.

Reply:
I'm not sure why the grammar treats relops separately from other binops. I think it might just be to assign different precedence to the relational operators?